packages/services/src/services/VideoPlayer/__tests__/VideoPlayer.test.js

```diff
@@ -0,0 +1,58 @@
+import apiDataResponse from './data/response.json';
+import root from 'window-or-global';
+import VideoPlayerAPI from '../VideoPlayer';
+
+let _scriptLoaded = false;
+
+const _partnerId = process.env.KALTURA_PARTNER_ID || 243342;
+const _uiConfId = process.env.KALTURA_UICONF_ID || 12905712;
+
+//jest.mock('../../VideoPlayer');
+
+describe('VideoPlayer', () => {
+  beforeEach(function() {
+    root.kWidget = {
+      api: {
+        doRequest: jest.fn(() => Promise.resolve(apiDataResponse)),
+      },
+      embed: jest.fn(() =>
+        Promise.resolve({
+          targetId: 'kaltura_player',
+          wid: '_' + _partnerId,
+          uiconf_id: _uiConfId,
+          entry_id: '0_uka1msg4',
+          flashvars: {
+            autoPlay: false,
+          },
+        })
+      ),
+    };
+  });
+
+  it('should set a loop to check script state is the loaded state or loading state', () => {
+    jest.useFakeTimers();
+    VideoPlayerAPI.checkScript();
```

Comment:
don't call `checkScript`, your tests should just be around the `embed` and `api` calls. `checkScript` should just be a side effect of testing those methods.